packages/browser/core/src/core/auth/internal/StitchAuthImpl.ts

```diff
@@ -294,6 +324,106 @@ export default class StitchAuthImpl extends CoreStitchAuth<StitchUser>
     }
   }
 
+  /**
+   * Utility function used to force the compiler to enforce an exhaustive 
+   * switch statment in dispatchAuthEvent at compile-time.
+   * @see https://www.typescriptlang.org/docs/handbook/advanced-types.html
+   */
+  private assertNever(x: never): never {
+    throw new Error("unexpected object: " + x);
+  }
+
+  /**
+   * Dispatch method for the new auth listener methods.
+   * @param event the discriminated union representing the auth event
+   */
+  public dispatchAuthEvent(event: AuthEvent<StitchUser>) {
+    switch(event.kind) {
+      case AuthEventKind.ActiveUserChanged:
+        this.dispatchBlockToListeners((listener: StitchAuthListener) => {
+          if (listener.onActiveUserChanged) {
+            listener.onActiveUserChanged(
+              this,
+              event.currentActiveUser,
+              event.previousActiveUser
+            );  
+          }
+        });
+        break;
+      case AuthEventKind.ListenerRegistered:
+        this.dispatchBlockToListeners((listener: StitchAuthListener) => {
+          if (listener.onListenerRegistered) {
+            listener.onListenerRegistered(this);  
+          }
+        });
+        break;
+      case AuthEventKind.UserAdded:
+        this.dispatchBlockToListeners((listener: StitchAuthListener) => {
+          if (listener.onUserAdded) {
+            listener.onUserAdded(this, event.addedUser);  
```

Comment:
So Typescript knows that `event` has added user, because this `event` is under the switch case `AuthEventKind.UserAdded`.

See "Discriminated Unions" in https://www.typescriptlang.org/docs/handbook/advanced-types.html